### PR TITLE
Add dates and changelog for packages and vulnerabilities

### PR DESCRIPTION
**Pull request by TG1999**

### CODE REVIEW

**Comment by TG1999** [comment]
@tdruez @johnmhoran @pombredanne please review this

File: vulnerabilities/models.py [hunk 1]
```diff
@@ -1098,3 +1149,192 @@ class ApiUser(UserModel):
...
+        blank=True,
+        help_text="VCID of the vulnerability related to this package",
+    )
+
+    action_type = models.PositiveSmallIntegerField(choices=ACTION_TYPE_CHOICES)
+
+    objects = PackageHistoryManager()
+
+    @classmethod
+    def log_affected_by(cls, package, importer, source_url, related_vulnerability):
```
Comment (keshav-space) [inline vulnerabilities/models.py:1317]: Do we not want to keep Import history at package level?

File: vulnerabilities/models.py [hunk 2]
```diff
@@ -1098,3 +1149,192 @@ class ApiUser(UserModel):
...
+            action_type=PackageChangeLog.AFFECTED_BY,
+            actor_name=importer,
+            source_url=source_url,
+            related_vulnerability=related_vulnerability,
+        )
+
+    @classmethod
+    def log_fixing(cls, package, importer, source_url, related_vulnerability):
+        """
+        Creates History entry on Vulnerabilitty is fixed by package.
```
Comment (keshav-space) [inline vulnerabilities/models.py:1332]:
```suggestion
        Creates History entry on Vulnerability is fixed by package.
```

File: vulnerabilities/models.py (hunk 1 above)
Comment (TG1999) [reply vulnerabilities/models.py:1317]: @keshav-space no, we just want to keep affected, fixed by or improvements on package, not the import logs.

Comment (keshav-space) [reply vulnerabilities/models.py:1317]: ack, in that case should we remove `IMPORT` as one of the action type choice in PackageChangeLog?

Comment (TG1999) [reply vulnerabilities/models.py:1316]: Done!

File: vulnerabilities/models.py [hunk 3]
```diff
@@ -1101,3 +1152,191 @@ class ApiUser(UserModel):
...
+
+    software_version = models.CharField(
+        max_length=100,
+        help_text="Version of the software at the time of change",
+        default=VULNERABLECODE_VERSION,
+    )
+
+    @property
+    def get_action_type_label(self):
+        label_by_status = {choice[0]: choice[1] for choice in self.ACTION_TYPE_CHOICES}
```
Comment (TG1999) [inline vulnerabilities/models.py:1190]: ### 
```suggestion
        label_by_status = {choice_code: choice_label for choice_code, choice_label in self.ACTION_TYPE_CHOICES}
```

File: vulnerabilities/package_managers.py [hunk 4]
```diff
@@ -0,0 +1,747 @@
+#
```
Comment (TG1999) [inline vulnerabilities/package_managers.py:1]: Delete the file

File: vulnerabilities/utils.py [hunk 5]
```diff
@@ -538,3 +539,12 @@ def get_severity_range(severity_list):
     if not score_list:
         return
     return f"{min(score_list)} - {max(score_list)}"
+
+
+@dataclasses.dataclass
```
Comment (TG1999) [inline vulnerabilities/utils.py:544]: Remove this object.

File: vulnerabilities/views.py [hunk 6]
```diff
@@ -71,6 +73,17 @@ def get_queryset(self, query=None):
         return self.model.objects.search(query=query).with_package_counts()
 
 
+key_func = lambda log: (
```
Comment (TG1999) [inline vulnerabilities/views.py:76]: Remove this

**Review by TG1999 — COMMENTED** [review]
Some nits!

**Pull request merged**